### PR TITLE
Attachment component enhancements

### PR DESCRIPTION
**Pull request by domoscargin**
https://trello.com/c/tUd7Ck2w/275-5-replace-document-links-with-attachment-component-in-frontend-apps

This PR is a result of testing the Attachment component in real-life settings.

It adds the following enhancements:

* A custom tag for the link text - this defaults to `h2`
* Allows the "description" property to be either text or html
* Adds a "last updated" property to the available metadata properties
* Adds a slimline version of the component with a small thumbnail, for use in environments like table rows where space is at more of a premium. It's also still possible to use the text-only version of the component, but that's probably more suitable for use within bodies of text.
* Fixes a bug where the accessible format request details component would display even if no details had been specified.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -9044,25 +9044,25 @@
       "dependencies": {
         "abbrev": {
           "version": "1.1.1",
-          "resolved": false,
+          "resolved": "https://registry.npmjs.org/abbrev/-/abbrev-1.1.1.tgz",
```
Comment (lfdebrux) [inline package-lock.json:9047]: Not sure why this change is here?

Comment (domoscargin) [reply package-lock.json:9047]: To be honest, nor am I - this is the result of `nvm use` `npm install`. Now that I've finished up, I'll blast the entire node_modules directory and try again - that sometimes helps.

File: src/digitalmarketplace/components/attachment/template.njk [hunk 2]
```diff
@@ -15,6 +15,9 @@
   {%- endif -%}
 {% endset %}
 
+{% set isThumbnailSmall = params.thumbnailSize and params.thumbnailSize == 'small' %}
```
Comment (lfdebrux) [inline src/digitalmarketplace/components/attachment/template.njk:18]: You could probably reduce some repetition later on by doing

```suggestion
{% set thumbnailClasses = " dm-attachment__thumbnail--small" if params.thumbnailSize == "small" %}
```

File: src/digitalmarketplace/components/attachment/template.njk [hunk 3]
```diff
@@ -68,9 +71,12 @@
           {%- if params.numberOfPages -%}
             , <span class="dm-attachment__attribute">{{ params.numberOfPages }} {% if params.numberOfPages == 1 %}page{% else %}pages{% endif %}</span>
           {% endif -%}
+          {%- if params.lastUpdated -%}
+            , <span class="dm-attachment__attribute">Last updated: <time datetime="{{ params.lastUpdated.datetime }}">{{ params.lastUpdated.text }}</time></span>
```
Comment (lfdebrux) [inline src/digitalmarketplace/components/attachment/template.njk:75]: What is this 😮 ? A new tag? https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time

The compatibility table suggests IE support is a no 🤔 I'm guess it just appears as text though?

File: package-lock.json (hunk 1 above)
Comment (domoscargin) [reply package-lock.json]: Yeah, blowing up `node_modules` and these changes and reinstalling did nothing, so I've wiped this commit.

File: src/digitalmarketplace/components/attachment/template.njk (hunk 3 above)
Comment (domoscargin) [reply src/digitalmarketplace/components/attachment/template.njk:75]: Yup - we've been happily using it on the Supplier Frontend from 4 years ago: https://github.com/alphagov/digitalmarketplace-supplier-frontend/blame/c8fca9349949d19fe187938bae7b911b83c0aced/app/templates/frameworks/_guidance_links.html

If a browser isn't familiar with it, it'll just treat it as a generic inline element. I've tested on Browserstack and it looks fine:

Screenshot 2020-11-30 at 10 22 52

Comment (lfdebrux) [reply src/digitalmarketplace/components/attachment/template.njk:75]: Huh, I had no idea 😅 So cool!

**Review by lfdebrux — APPROVED** [review]
